libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/component/marker/MapMarkersComponent.kt

```diff
@@ -31,8 +28,15 @@ internal open class MapMarkersComponent(
             .observe { point ->
                 annotationManager.deleteAll()
                 if (point != null) {
-                    val annotation = mapMarkerFactory.createPin(point, iconImage)
-                    annotationManager.create(annotation)
+                    annotationManager.create(
+                        markerAnnotationOptions.apply {
+                            if (markerAnnotationOptions.getPoint() == null ||
+                                markerAnnotationOptions.getPoint() != point
+                            ) {
+                                withPoint(point)
+                            }
```

Comment:
do we need `else` branch?

Reply:
My thinking here is:
- If you are injecting a custom `markerAnnotationOption`, you would either be setting the point or not. 
      - If you don't set a point then `getPoint` returns null and we add it.
      - If you set a point but that is stale while the `point` from Drop-In UI is `fresh` we add the new point
- The only use case left is `markerAnnotationOptions.getPoint() != null` && `markerAnnotationOptions.getPoint() == point` in which case the point is already set to `markerAnnotationOptions` and we can use it.

Do you think I missed something else? @korshaknn

Reply:
Looks like we can simply call `withPoint(point)` and not check anything.